administration/troubleshooting.md

```diff
@@ -0,0 +1,311 @@
+# Troubleshooting
+
+* [Tap Functionality: generate events or records](troubleshooting.md#tap-functionality)
+* [Dump Internals Signal](troubleshooting.md#dump-internals--signal)
+
+## Tap Functionality
+
+Tracing can be used to generate events or records detailing what messages
+pass through fluent-bit, at what time and what filters affect them.
+
+To test out tracing we will be using a publicly accessible docker container: 'fluent/fluent-bit:latest
+'.
+
+### Simple example
+
+First we will make sure that the container we are going to use actually supports tracing:
+
+```shell
+$ docker run --rm -ti ghcr.io/calyptia/core/calyptia-fluent-bit:main --help | grep -Z
+  -Z, --enable-chunk-trace     enable chunk tracing. activating it requires using the HTTP Server API.
+```
+
+If the `--enable-chunk-trace` option is present it means fluent-bit has tracing disabled by default, so has to be enabled when writing command.
+
+Tracing support is enabled and disabled via the embedded API server, so to enable it while fluent-bit is running it must be turned on:
+
+```shell
+$ docker run --rm -ti -p 2020:2020 ghcr.io/calyptia/core/calyptia-fluent-bit:main -Z -H -i dummy -p alias=input_dummy -o stdout -f 1
+Calyptia Fluent Bit v22.8, patch_level=1
+* Enterprise Fluent Bit by Calyptia
+* https://calyptia.com
+* https://fluentbit.io
+
+[2022/07/27 20:02:04] [ info] [calyptia fluent bit] version=22.8 patch=1 pid=1
+[2022/07/27 20:02:04] [ info] [storage] version=1.2.0, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
+[2022/07/27 20:02:04] [ info] [cmetrics] version=0.3.5
+[2022/07/27 20:02:04] [ info] [output:stdout:stdout.0] worker #0 started
+[2022/07/27 20:02:04] [ info] [http_server] listen iface=0.0.0.0 tcp_port=2020
+[2022/07/27 20:02:04] [ info] [sp] stream processor started
+[0] dummy.0: [1658952125.096215620, {"message"=>"dummy"}]
+[0] dummy.0: [1658952126.096222898, {"message"=>"dummy"}]
+[0] dummy.0: [1658952127.096220305, {"message"=>"dummy"}]
+...
+```
+
+In another window we can activate tracing by either using the instance id of the input; `dummy.0` or its alias. Since the alias is more predictable that is what we will use:
```

Comment:
```suggestion
In another window we can activate Tap by either using the instance id of the input; `dummy.0` or its alias. Since the alias is more predictable that is what we will use:
```